### PR TITLE
LST-1159 Implement cancellation of bookings.

### PR DESCRIPTION
**Pull request by oskusalerma**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,19 @@ Go to http://localhost:8000 and use your DIT Google account to log in.
...
 requirements files: `make build`.
 
+### Making yourself superuser
+
+After you login for the first time, do this to give yourself superuser access:
+
+```
+$ make shell
+import custom_usermodel
+me = custom_usermodel.models.User.objects.all()[0]
```
Comment (lgarvey) [inline README.md:56]: minor, but.. .

`User.objects.first()` is a bit more direct

File: main/models.py [hunk 2]
```diff
@@ -40,3 +43,24 @@ class Booking(models.Model):
...
+    def get_on_behalf_of(self):
+        """Return a string describing who the booking is on behalf of. It will be one of:
+
+        "Yourself"
+        "Donald Duck"
+        "donald.duck@digital.trade.gov.uk"
+        "Donald Duck (donald.duck@digital.trade.gov.uk)"
+        """
+
+        if self.on_behalf_of_name or self.on_behalf_of_dit_email:
```
Comment (rossmiller) [inline main/models.py:57]: You could do:

```
if not self.on_behalf_of_name and not self.on_behalf_of_dit_email:
    return "Yourself"

return f"{self.on_behalf_of_name}{f' ({self.on_behalf_of_dit_email})' if self.on_behalf_of_dit_email else ''}"
```
but I am not sure if it's easier to read or not.

**Review by rossmiller — APPROVED** [review]
Looks good! I left a comment about a possible to change to the conditionals in `get_on_behalf_of`.

File: main/models.py (hunk 2 above)
Comment (oskusalerma) [reply main/models.py:57]: That's missing one of the four cases, doesn't pass the tests:

```
main/tests/test_models.py:11: in test_booking_on_behalf_of_dit_email
    assert b.get_on_behalf_of() == "jane@blaa.com"
E   AssertionError: assert 'None (jane@blaa.com)' == 'jane@blaa.com'
E     - jane@blaa.com
E     + None (jane@blaa.com)
E     ? ++++++             +

```

**Pull request merged**